scrapper.py

```diff
@@ -26,38 +36,125 @@ class Crawler:
         """
         Returns seed_urls param
         """
-        pass
-
+        return self.seed_urls
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def parse(self):
+        response = requests.get(self.article_url, headers=HEADERS)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        page_link = article_bs.find('ul', class_='value galleys_links').find('a')['href']
+        response_pdf = requests.get(page_link, headers=HEADERS)
+        pdf_bs = BeautifulSoup(response_pdf.text, 'html.parser')
+        download_link = pdf_bs.find('header', class_='header_view').find('a', class_='download')['href']
+        pdf = PDFRawFile(download_link, self.article_id)
+        pdf.download()
+        self.article.text = pdf.get_text()
+
+    def _fill_article_with_meta_information(self, article_bs):
+        article_title = article_bs.find('h1', class_='page_title').text
+        article_title = article_title.strip()
+        self.article.title = article_title
+        article_author = article_bs.find('ul', class_='item authors')
+        article_author = article_author.find('span', class_='name')
+        if not article_author:
+            article_author = 'NOT FOUND'
+        else:
+            article_author = article_author.text
+            article_author = article_author.strip()
+        self.article.author = article_author
+        article_date = article_bs.find('div', class_='item published').find('div', class_='value')
+        article_date = article_date.text.strip()
+        article_date = datetime.strptime(article_date, '%Y-%m-%d')
+        self.article.date = article_date
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
-
+    path_to_base_path = Path(base_path)
+    try:
+        shutil.rmtree(base_path)
+    except FileNotFoundError:
+        pass
+    path_to_base_path.mkdir(parents=True)
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as file:
+        scrapper_config = json.load(file)
+
+    seed_urls = scrapper_config["seed_urls"]
+    max_articles = scrapper_config["total_articles_to_find_and_parse"]
+
+    if 'total_articles_to_find_and_parse' not in scrapper_config:
+        raise IncorrectNumberOfArticlesError
+
+    if 'seed_urls' not in scrapper_config:
+        raise IncorrectURLError
+
+    if not seed_urls or not isinstance(seed_urls, list):
+        raise IncorrectURLError
+
+    for seed_url in seed_urls:
+        if not isinstance(seed_url, str):
```

Comment:
this is a very naive check, any string can be a valid URL? you need to find more specific rules for detecting URLs and even more specific for your particular website as you for sure do not want to allow anyone give you the URL like this: https://github.com/fipl-hse/2021-2-level-ctlr/pull/23/files resulting in your program crash